esphome/components/http_request/http_request.cpp

```diff
@@ -20,5 +20,27 @@ void HttpRequestComponent::dump_config() {
   }
 }
 
+std::string HttpContainer::get_response_header(const std::string &header_name) {
+  ESP_LOGD(TAG, "Entered get_response_header");
+  auto response_headers = this->get_response_headers();
+  auto header_name_lower_case = str_lower_case(header_name);
+  if (response_headers.count(header_name_lower_case) == 0) {
+    ESP_LOGD(TAG, "No header with name %s found", header_name_lower_case.c_str());
+    return "";
+  } else {
+    auto values = response_headers[header_name_lower_case];
+    if (values.empty()) {
+      ESP_LOGD(TAG, "header with name %s returned an empty list, this shouldn't happen",
```

Comment:
```suggestion
      ESP_LOGE(TAG, "header with name %s returned an empty list, this shouldn't happen",
```
If this shouldn't happen, the log should probably be an Error log then, or at least a warning